### PR TITLE
*: rename storage package to mvcc

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

**Comment by gyuho** [comment]
sorry, could you rebase after this https://github.com/coreos/etcd/pull/5188?

**Comment by heyitsanthony** [comment]
@gyuho ok all rebased

**Comment by gyuho** [comment]
LGTM. Thanks. /cc @xiang90

**Comment by xiang90** [comment]
LGTM

**Comment by heyitsanthony** [comment]
merging on greenlight

File: raft/raft.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ type Config struct {
 	HeartbeatTick int
 
 	// Storage is the storage for raft. raft generates entries and states to be
-	// stored in storage. raft reads the persisted entries and states out of
+	// stored in mvcc. raft reads the persisted entries and states out of
```
Comment (xiang90) [inline raft/raft.go]: change this back to storage?

File: raft/rafttest/node.go [hunk 2]
```diff
@@ -69,9 +69,9 @@ func (n *node) start() {
 			case rd := <-n.Ready():
 				if !raft.IsEmptyHardState(rd.HardState) {
 					n.state = rd.HardState
-					n.storage.SetHardState(n.state)
+					n.mvcc.SetHardState(n.state)
```
Comment (xiang90) [inline raft/rafttest/node.go]: this should be still storage?

File: wal/wal.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ var (
 	crcTable            = crc32.MakeTable(crc32.Castagnoli)
 )
 
-// WAL is a logical representation of the stable storage.
+// WAL is a logical representation of the stable mvcc.
```
Comment (xiang90) [inline wal/wal.go]: mvcc -> storage

File: etcdserver/storage.go [hunk 4]
```diff
@@ -30,10 +30,10 @@ import (
 )
 
 type Storage interface {
-	// Save function saves ents and state to the underlying stable storage.
-	// Save MUST block until st and ents are on stable storage.
+	// Save function saves ents and state to the underlying stable mvcc.
```
Comment (xiang90) [inline etcdserver/storage.go]: this should be changed back to storage.

**Pull request merged**